### PR TITLE
Get rid of admin-configurable question paths.

### PR DESCRIPTION
**Pull request by kendrickt**
Question path is now defined by a Question Parent Path and the Question Name.

### Description

In `QuestionForm`, add `questionParentPath` and drop `questionPath`. In `QuestionForm.getBuilder()`, the `QuestionDefinition.questionPath` is now set by concatenating the parent path and a formatted `questionName`. 

The `QuestionEditView` no longer use a text field for `question path`. Instead, a dropdown selector for `questionParentPath` is added. It just has "Applicant" for now, but paths for repeated elements will go here in the future.

Tests were updated (mostly browser tests).

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)

image

### CODE REVIEW

File: universal-application-tool-0.0.1/test/app/ApplicantProgramBrowserTest.java [hunk 1]
```diff
@@ -26,9 +26,9 @@ public void setUp() {
 
     loginAsAdmin();
 
-    addNameQuestion("name", "applicant.name");
-    addTextQuestion("color", "applicant.color");
-    addAddressQuestion("address", "applicant.address");
+    addNameQuestion("name");
+    addTextQuestion("color");
+    addAddressQuestion("address");
```
Comment (AWBaumann) [inline universal-application-tool-0.0.1/test/app/ApplicantProgramBrowserTest.java:31]: Nice! These lines really show what's going on in the PR.

**Review by AWBaumann — APPROVED** [review]
Nice work!

File: universal-application-tool-0.0.1/app/forms/QuestionForm.java [hunk 2]
```diff
@@ -63,12 +63,18 @@ public void setQuestionDescription(String questionDescription) {
...
-  public void setQuestionPath(String questionPath) {
-    this.questionPath = Path.create(checkNotNull(questionPath));
+  public void setQuestionParentPath(String questionParentPath) {
+    this.questionParentPath = Path.create(checkNotNull(questionParentPath));
+  }
+
+  public Path getQuestionPath() {
+    String questionNameFormattedForPath =
+        questionName.replaceAll("\\s", "_").replaceAll("[^a-zA-Z_]", "");
+    return questionParentPath.with(questionNameFormattedForPath);
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/forms/QuestionForm.java:77]: nit: feel free to disagree, but I feel like `with` could be renamed to `append`

File: universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java [hunk 3]
```diff
@@ -173,6 +167,20 @@ private ContainerTag buildQuestionForm(QuestionForm questionForm) {
...
 
+  private DomContent questionParentPathSelect() {
+    // TODO: add repeated element paths when they exist.
+    ImmutableList<SimpleEntry<String, String>> options =
+        ImmutableList.of(new SimpleEntry<>("Applicant", "applicant"));
+
+    return new SelectWithLabel()
+        .setId("question-parent-path-select")
+        .setFieldName("questionParentPath")
+        .setLabelText("Question parent path")
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java:178]: suggestion (non-blocking): Perhaps rename this field to just "Question parent"?

File: universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java [hunk 4]
```diff
@@ -173,6 +167,20 @@ private ContainerTag buildQuestionForm(QuestionForm questionForm) {
     return formTag;
   }
 
+  private DomContent questionParentPathSelect() {
+    // TODO: add repeated element paths when they exist.
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java:171]: nit: add issue for this TODO - perhaps https://github.com/seattle-uat/civiform/issues/405?

**Review by carolinedanzi — APPROVED** [review]
Exciting progress!

File: universal-application-tool-0.0.1/app/forms/QuestionForm.java (hunk 2 above)
Comment (kendrickt) [reply universal-application-tool-0.0.1/app/forms/QuestionForm.java:77]: I'm gonna use "join", from python's `os.path.join` and node's `path.join`.

File: universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java (hunk 4 above)
Comment (kendrickt) [reply universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java:171]: done

File: universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java (hunk 3 above)
Comment (kendrickt) [reply universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java:178]: It contains the parent's path, so I think it is clearer to keep it named path.

**Pull request merged**